packages/snap-preact-components/src/components/Organisms/BundledRecommendations/BundledRecommendations.tsx

```diff
@@ -0,0 +1,515 @@
+		seperatorIcon: 'plus-thin',
+		seedText: 'Seed Product',
+		loop: false,
+		addToCartText: 'Add All To Cart',
+		mobileMediaQuery: '(max-width: 650px)',
+		// global theme
+		...globalTheme?.components?.recommendation,
+		...properties,
+		// props
+		...properties.theme?.components?.recommendation,
+	};
+
+	const displaySettings = useDisplaySettings(props.breakpoints!);
+	if (displaySettings && Object.keys(displaySettings).length) {
+		const theme = deepmerge(props?.theme || {}, displaySettings?.theme || {}, { arrayMerge: (destinationArray, sourceArray) => sourceArray });
+		props = {
+			...props,
+			...displaySettings,
+			theme,
+		};
+	}
+
+	const {
+		title,
+		controller,
+		breakpoints,
+		loop,
+		results,
+		seedInCarousel,
+		pagination,
+		nextButton,
+		prevButton,
+		hideButtons,
+		preselectedCount,
+		seperatorIcon,
+		showCheckboxes,
+		seedText,
+		showQuantityPicker,
+		onAddToCart,
+		seedIconOnly,
+		resultComponent,
+		ctaSlot,
+		addToCartText,
+		disableStyles,
+		peekabooEnableAt,
+		style,
+		mobileMediaQuery,
+		className,
+		...additionalProps
+	} = props;
+
+	if (!controller || controller.type !== 'recommendation') {
+		throw new Error(`<Recommendation> Component requires 'controller' prop with an instance of RecommendationController`);
+	}
+
+	//need to make a clone of products to show to prevent infinite re-renders when changing selected items
+	const products = results || controller.store?.results;
+	const resultsToRender: Product[] = [...products];
+
+	const seed = results ? results[0] : controller.store?.results[0];
+
+	const subProps: RecommendationSubProps = {
+		carousel: {
+			loop: loop,
+			// default props
+			className: 'ss__recommendation__carousel',
+			// global theme
+			...globalTheme?.components?.carousel,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+		result: {
+			// default props
+			className: 'ss__recommendation__result',
+			// global theme
+			...globalTheme?.components?.result,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+	};
+
+	const isMobile = useMediaQuery(mobileMediaQuery!);
+
+	const slidesPerView = props.slidesPerView;
+	const styling: { css?: StylingCSS } = {};
+	if (!disableStyles) {
+		styling.css = [CSS.bundledRecommendations({ slidesPerView, isMobile }), style];
+	} else if (style) {
+		styling.css = [style];
+	}
+
+	const modded: any = { ...breakpoints };
+
+	const initialSelectedIds: string[] = [];
+
+	const _preSelectedCount = typeof preselectedCount == 'number' ? preselectedCount : slidesPerView;
+
+	if (_preSelectedCount) {
+		resultsToRender?.forEach((result, idx) => {
+			if (idx < _preSelectedCount) {
+				initialSelectedIds.push(result.id);
+			}
+		});
+	}
+
+	const [selectedIds, setSelectedIds] = useState<string[]>(initialSelectedIds);
+
+	const selectedItems = selectedIds.map((id) => ({
```

Comment:
After thinking about this more, I think it makes more sense to have this be an array of products (a reference to the full product from the store). This works better for a few reasons:

- future proof (all properties are available)
- the quantity is already in the product store
- consistency - not adding some additional object (in this case one with simply `id` and `quantity`

Reply:
I like this more.